cloudinit/sources/DataSourceNoCloud.py

```diff
@@ -357,6 +358,30 @@ def __init__(self, sys_cfg, distro, paths):
         DataSourceNoCloud.__init__(self, sys_cfg, distro, paths)
         self.supported_seed_starts = ("http://", "https://")
 
+    def ds_detect(self):
+        """NoCloud requires "nocloud-net" as the way to specify
+        seeding from an http(s) address. This diverges from all other
+        datasources in that it does a kernel commandline match on something
+        other than the datasource dsname for only DEP_NETWORK.
+        """
+        cmdline = util.get_cmdline()
+        ds_parse_0 = re.search(r"ds=([a-zA-Z]+)(\s|$)", cmdline)
+        ds_parse_1 = re.search(r"ci\.ds=([a-zA-Z]+)(\s|$)", cmdline)
+        ds_parse_2 = re.search(r"ci\.datasource=([a-zA-Z]+)(\s|$)", cmdline)
+        ds = ds_parse_0 or ds_parse_1 or ds_parse_2
+        deprecated = ds_parse_1 or ds_parse_2
+        if deprecated:
+            util.deprecate(
+                deprecated=(
+                    f"Defining the datasource on the commandline using "
+                    f"ci.ds={ds} or ci.datasource={ds}"
+                ),
+                deprecated_version="23.2",
+                extra_message=f"Use ds={ds} instead",
```

Comment:
Again printing re.search results.
```suggestion
                extra_message=f"Use ds={ds.group(1)} instead",
```